### PR TITLE
ci: Add a workflow for creating releases

### PR DESCRIPTION
**Pull request by stephenmathieson**
Example release PR: https://github.com/dequelabs/axe-core/pull/3814

Ref https://github.com/dequelabs/axe-api-team/issues/304

<< Describe the changes >>

Closes issue:

### CODE REVIEW

**Review by WilcoFiers — DISMISSED** [review]
LGTM, I would like @straker's review of this too before we merge.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Run release script and open PR
+        run: |
+          git config user.name "API Team CI User"
+          git config user.email "aciattestteamci@deque.com"
+
+          Branch="release-$(date +%Y-%m-%d)"
+          echo "Branch: $Branch"
+          git checkout -b "$Branch"
+
+          npm ci
```
Comment (straker) [inline .github/workflows/release.yml:29]: The release [already runs `npm ci`](https://github.com/dequelabs/axe-core/blob/develop/package.json#L60-L62) so no need to run it before

```suggestion
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          npm run release
+
+          git push origin "$Branch" --force
```
Comment (straker) [inline .github/workflows/release.yml:32]: Why the force push?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+          Version=$(jq -r .version ./package.json)
+          echo "Version: $Version"
+
+          # Stolen from https://github.com/dequelabs/axe-extension/blob/0f22963c4d0a73bf853aa982ade2393e422f002c/scripts/release.sh#L51-L54
```
Comment (straker) [inline .github/workflows/release.yml:37]: This is a private url, shouldn't list it in public repo.

```suggestion
          # Get the additions to the changelog as the commit body and generate the PR url
```

**Review by straker — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (stephenmathieson) [reply .github/workflows/release.yml:31]: If the branch already has changes in it, we'll want to overwrite them.

File: .github/workflows/release.yml (hunk 3 above)
Comment (stephenmathieson) [reply .github/workflows/release.yml:36]: Fair enough 🤷

File: .github/workflows/release.yml (hunk 2 above)
Comment (straker) [reply .github/workflows/release.yml:31]: Wouldn't the checkout action fail then and kill the build before it got to this point?

Comment (stephenmathieson) [reply .github/workflows/release.yml:31]: No, the checkout action is taking whatever branch we're releasing from (usually `develop`). We then create this branch, add the release commit, then push. The `--force` is to ensure any prior pending release (maybe someone made a mistake) is overwritten.

This is, however, fixing a problem we haven't had (yet?). I'm happy to pull it out if you aren't comfortable force pushing.

Comment (straker) [reply .github/workflows/release.yml:31]: Wilco didn't seem to think it was a problem when he reviewed it, so I'm fine with it.

**Review by straker — APPROVED** [review]

**Review by WilcoFiers — APPROVED** [review]

**Pull request merged**